roles/utils/kerberos/tasks/check_secure_keytab.yml

```
@@ -0,0 +1,16 @@
+# Copyright 2022 TOSIT.IO
+# SPDX-License-Identifier: Apache-2.0
+
+---
+- name: Check kinit for {{ principal }}
+  shell: |
+    kinit -kt {{ keytabs_dir }}/{{ keytab }} {{ principal }}@{{ realm }}
+    klist | grep "Default principal: {{ principal }}@{{ realm }}"
```

I disagree with the check. If the `root` user already have a TGT you will destroy it with the `kinit`. In order to not override an existing TGT you should change the Kerberos credentials cache location.

```suggestion
    kinit -kt {{ keytabs_dir }}/{{ keytab }} {{ principal }}@{{ realm }} -c /tmp/check_keytab_cache
    klist /tmp/check_keytab_cache | grep "Default principal: {{ principal }}@{{ realm }}"
    rm /tmp/check_keytab_cache
```

---

Agreed on that. Nice catch! Will apply the required changes.